src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsPublisher.cs

```diff
@@ -194,33 +194,41 @@ internal async Task PublishDiagnosticsAsync(DocumentSnapshot document)
         private async void WorkTimer_Tick(object state)
 #pragma warning restore VSTHRD100 // Avoid async void methods
         {
-            DocumentSnapshot[] documents;
-            lock (_work)
-            {
-                documents = _work.Values.ToArray();
-                _work.Clear();
-            }
-
-            for (var i = 0; i < documents.Length; i++)
+            try
             {
-                var document = documents[i];
-                await PublishDiagnosticsAsync(document);
-            }
+                DocumentSnapshot[] documents;
+                lock (_work)
+                {
+                    documents = _work.Values.ToArray();
+                    _work.Clear();
+                }
 
-            OnCompletingBackgroundWork();
+                for (var i = 0; i < documents.Length; i++)
+                {
+                    var document = documents[i];
+                    await PublishDiagnosticsAsync(document);
+                }
 
-            lock (_work)
-            {
-                // Resetting the timer allows another batch of work to start.
-                _workTimer.Dispose();
-                _workTimer = null;
+                OnCompletingBackgroundWork();
 
-                // If more work came in while we were running start the timer again.
-                if (_work.Count > 0)
+                lock (_work)
                 {
-                    StartWorkTimer();
+                    // Resetting the timer allows another batch of work to start.
+                    _workTimer.Dispose();
+                    _workTimer = null;
+
+                    // If more work came in while we were running start the timer again.
+                    if (_work.Count > 0)
+                    {
+                        StartWorkTimer();
+                    }
                 }
             }
+            catch (Exception ex)
+            {
+                // Swallow exception to not crash VS (this is an async void method)
```

Comment:
If something throws while in an async-void method it'll crash the parent process